src/scripts/mutual_checker.js

```diff
@@ -50,72 +74,103 @@ const addIcons = function (postElements) {
     const blogName = postAttribution.textContent.trim();
     if (!blogName) return;
 
-    if (following[blogName] === undefined) {
-      const { blog } = await timelineObject(postElement);
-      if (blogName === blog.name) {
-        following[blogName] = Promise.resolve(blog.followed && !blog.isMember);
-      } else {
-        following[blogName] = apiFetch(`/v2/blog/${blogName}/info`)
-          .then(({ response: { blog: { followed } } }) => followed)
-          .catch(() => Promise.resolve(false));
-      }
-    }
-
-    const followingBlog = await following[blogName];
+    const followingBlog = await getIsFollowing(blogName, postElement);
     if (!followingBlog) { return; }
 
-    if (mutuals[blogName] === undefined) {
-      mutuals[blogName] = apiFetch(`/v2/blog/${primaryBlogName}/followed_by`, { queryParams: { query: blogName } })
-        .then(({ response: { followedBy } }) => followedBy)
-        .catch(() => Promise.resolve(false));
-    }
-
-    const isMutual = await mutuals[blogName];
+    const isMutual = await getIsFollowingYou(blogName);
     if (isMutual) {
       postElement.classList.add(mutualsClass);
       const iconTarget = getPopoverWrapper(postAttribution) ?? postAttribution;
-      iconTarget?.before(icon.cloneNode(true));
+      iconTarget?.before(createIcon(blogName));
     } else if (showOnlyMutuals) {
       getTimelineItemWrapper(postElement)?.setAttribute(hiddenAttribute, '');
     }
   });
 };
 
+const addBlogCardIcons = blogCardLinks =>
+  blogCardLinks.forEach(async blogCardLink => {
+    const blogName = blogCardLink.querySelector(keyToCss('blogLinkShort'))?.textContent || blogCardLink?.textContent;
+    if (!blogName) return;
+
+    const followingBlog = await getIsFollowing(blogName, blogCardLink);
+    if (!followingBlog) return;
+
+    const isMutual = await getIsFollowingYou(blogName);
+    if (isMutual) {
+      blogCardLink.before(createIcon(blogName, blogCardLink.style.color));
```

Comment:
Ah, I see. Well, we can fix this by using `getComputedStyle` instead, which will always return a valid value...

```suggestion
      blogCardLink.before(createIcon(blogName, getComputedStyle(blogCardLink).color));
```

...or is `getComputedStyle` too expensive? 😛

Reply:
I don't think I was worried about it not working! But this is more resilient, so why not.